bip-musig2/reference.py

```diff
@@ -377,18 +477,82 @@ def test_sign_vectors():
+        '037496A3CC86926D452CAFCFD55D25972CA1675D549310DE296BFF42F72EEEA8C9')
+    session_ctx = SessionContext(invalid_aggnonce, [X[0], X[1], pk], [], [], msg)
+    assertRaises(InvalidContributionError,
+                 lambda: sign(secnonce, sk, session_ctx),
+                 lambda e: e.signer == None)
+    # Vector 6: Aggregate nonce is invalid because the second half does not
+    # correspond to an X coordinate
+    invalid_aggnonce = bytes.fromhex(
+        '028465FCF0BBDBCF443AABCCE533D42B4B5A10966AC09A49655E8C42DAAB8FCD61' +
+        '020000000000000000000000000000000000000000000000000000000000000009')
+    session_ctx = SessionContext(invalid_aggnonce, [X[0], X[1], pk], [], [], msg)
+    assertRaises(InvalidContributionError,
+                 lambda: sign(secnonce, sk, session_ctx),
+                 lambda e: e.signer == None)
+    # Vector 7: Aggregate nonce is invalid because second half exceeds field
+    # size
+    invalid_aggnonce = bytes.fromhex(
+        '028465FCF0BBDBCF443AABCCE533D42B4B5A10966AC09A49655E8C42DAAB8FCD61' +
+        '02FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFC30')
+    session_ctx = SessionContext(invalid_aggnonce, [X[0], X[1], pk], [], [], msg)
+    assertRaises(InvalidContributionError,
+                 lambda: sign(secnonce, sk, session_ctx),
+                 lambda e: e.signer == None)
+
+    # Verification test vectors
+    # Vector 8
+    assert partial_sig_verify(expected[0], pnonce, [pk, X[0], X[1]], [], [], msg, 0)
+    # Vector 9
+    assert partial_sig_verify(expected[1], [pnonce[1], pnonce[0], pnonce[2]], [X[0], pk, X[1]], [], [], msg, 1)
+    # Vector 10
+    assert partial_sig_verify(expected[2], [pnonce[1], pnonce[2], pnonce[0]], [X[0], X[1], pk], [], [], msg, 2)
+
+    # Vector 11: Wrong signature (which is equal to the negation of valid signature expected[0])
+    wrong_sig = bytes.fromhex('97AC833ADCB1AFA42EBF9E0725616F3C9A0D5B614F6FE283CEAAA37A8FFAF406')
+    assert not partial_sig_verify(wrong_sig, pnonce, [pk, X[0], X[1]], [], [], msg, 0)
+    # Vector 12: Wrong signer
+    assert not partial_sig_verify(expected[0], pnonce, [pk, X[0], X[1]], [], [], msg, 1)
+    # Vector 13: Signature exceeds group size
+    wrong_sig = bytes.fromhex('FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEBAAEDCE6AF48A03BBFD25E8CD0364141')
+    assert not partial_sig_verify(wrong_sig, pnonce, [pk, X[0], X[1]], [], [], msg, 0)
+    # Vector 14: Invalid pubnonce
+    invalid_pubnonce = bytes.fromhex('020000000000000000000000000000000000000000000000000000000000000009')
+    assertRaises(InvalidContributionError,
+                 lambda: partial_sig_verify(expected[0], [invalid_pubnonce, pnonce[1], pnonce[2]], [pk, X[0], X[1]], [], [], msg, 0),
+                 lambda e: e.signer == 0)
+    # Vector 15: Invalid public key
+    assertRaises(InvalidContributionError,
+                 lambda: partial_sig_verify(expected[0], [invalid_pk, pnonce[1], pnonce[2]], [pk, X[0], X[1]], [], [], msg, 0),
```

Comment:
Nit: Did you intend to do this instead (since you're currently putting the `invalid_pk` in the `pnonce[0]` position)?

```python
partial_sig_verify(expected[0], pnonce, [invalid_pk, X[0], X[1]], [], [], msg, 0),
```

Reply:
Oops, yes. Fixed